recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,369 @@
+        "loguru",
+        "lz4",
+        "lzma",
+        "metaio",
+        "netcdf",
+        "octree",
+        "ogg",
+        "opengl",
+        "pegtl",
+        "png",
+        "pugixml",
+        "sqlite",
+        "theora",
+        "tiff",
+        "utf8",
+        "verdict",
+        "vpic",
+        "vtkDICOM",
+        "vtkm",
+        "vtksys",
+        "xdmf2",
+        "xdmf3",
+        "zfp",
+        "zlib",
+    ]
+    options = dict({"shared": [True, False], "fPIC": [True, False],
+    }, **{"group_{}".format(group.lower()): [True, False] for group in _groups},
+    **{"module_{}".format(module.lower()): [True, False] for module in _modules}
+    )
+    # default_options are set to the same values as clean VTK 9.0.1 cmake installation has, except "shared" which Conan require to be "False" by default.
+    default_options = dict({
+        "shared": False,
+        "fPIC": False,
+        }, **{"group_{}".format(group.lower()): True for group in _groups if (group in ["StandAlone", "Rendering"])},
+        **{"group_{}".format(group.lower()): False for group in _groups if (group not in ["StandAlone", "Rendering"])},
+        **{"module_{}".format(module.lower()): False for module in _modules})
+    short_paths = True
+    _source_subfolder = "source_subfolder"
+    _cmake = None
+
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler = str(self.settings.compiler)
+        if compiler == "apple-clang" and self.options.shared:
+            raise ConanInvalidConfiguration("VTK on apple-clang requires to be static (vtk:shared=False).")
+        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime and self.options.shared:
+            raise ConanInvalidConfiguration("VTK can not be built as shared library with MT runtime.")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.upper() + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def requirements(self):
+        if self.options.group_rendering:
+            self.requires("opengl/system")
+        if self.options.module_ioxdmf3:
+            self.requires("boost/1.74.0")
+        if self.options.group_qt:
+            # FIXME: Missing qt recipe. Qt recipe PR: https://github.com/conan-io/conan-center-index/pull/1759
+            # When qt available, "self.requires("qt/5.15.1")" should replace below line.
+            raise ConanInvalidConfiguration("qt is not (yet) available on conan-center-index")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = "OFF"
+        self._cmake.definitions["BUILD_EXAMPLES"] = "OFF"
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = "ON" if self.options.shared else "OFF"
```

Comment:
Conan handles the translation

```suggestion
        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
```

Reply:
I see elsewhere it's very strict... is that the case here?